# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/Layers.h
# Commit: 16c5dd2cfd60
# Full Hash: 16c5dd2cfd600a62c80ca17cfb35c0868e016056
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r ca8139d58d58 -r 16c5dd2cfd60 gfx/layers/Layers.h
--- a/gfx/layers/Layers.h	Fri Jun 12 21:53:14 2020 +0000
+++ b/gfx/layers/Layers.h	Mon Jun 15 18:25:55 2020 +0000
@@ -77,7 +77,6 @@
 namespace layers {
 
 class Animation;
-class AsyncCanvasRenderer;
 class AsyncPanZoomController;
 class BasicLayerManager;
 class ClientLayerManager;
@@ -2540,7 +2539,7 @@
 
   const nsIntRect& GetBounds() const { return mBounds; }
 
-  CanvasRenderer* CreateOrGetCanvasRenderer();
+  RefPtr<CanvasRenderer> CreateOrGetCanvasRenderer();
 
  public:
   /**
@@ -2581,9 +2580,9 @@
   void DumpPacket(layerscope::LayersPacket* aPacket,
                   const void* aParent) override;
 
-  virtual CanvasRenderer* CreateCanvasRendererInternal() = 0;
+  virtual RefPtr<CanvasRenderer> CreateCanvasRendererInternal() = 0;
 
-  UniquePtr<CanvasRenderer> mCanvasRenderer;
+  RefPtr<CanvasRenderer> mCanvasRenderer;
   gfx::SamplingFilter mSamplingFilter;
 
   /**